static-analysis/EnumValidation.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MyCLabs\Tests\Enum\StaticAnalysis;
+
+use MyCLabs\Enum\Enum;
+
+/**
+ * @psalm-immutable
+ * @psalm-template T of 'A'|'C'
+ * @template-extends Enum<T>
+ */
+final class ValidationEnum extends Enum
+{
+    const A = 'A';
+    const C = 'C';
+}
+
+/**
+ * @psalm-pure
+ * @param mixed $input
+ * @psalm-return 'A'|'C'
+ *
+ * @psalm-suppress MixedReturnStatement
+ * @psalm-suppress MixedInferredReturnType at the time of this writing, we did not yet find
```

Comment:
This comment is kinda oudated - perhaps I can drop the static analysis check overall, now that I designed `Enum::assertValidEnumValue()` :thinking: